Kubexit fails when a timeout of birth dependencies is reached

Kubexit fails with an exception in the case a timeout is reached when waiting for the birth dependency.

Container log:
```
2020-07-29T14:39:59.848087360Z Name: example
2020-07-29T14:39:59.848148037Z Graveyard: /kubexit
2020-07-29T14:39:59.848152930Z Tombstone: /kubexit/<example>
2020-07-29T14:39:59.848156107Z Birth Deps: <birth-dep-example>
2020-07-29T14:39:59.848159006Z Death Deps: N/A
2020-07-29T14:39:59.848162280Z Birth Timeout: 30s
2020-07-29T14:39:59.848165060Z Grace Period: 30s
2020-07-29T14:39:59.848167959Z Pod Name: <pod-name-redacted>
2020-07-29T14:39:59.848171034Z Namespace: default
2020-07-29T14:39:59.848200329Z Watching pod updates...
2020-07-29T14:40:29.848407661Z Error: timed out waiting for birth deps to be ready: 30s
2020-07-29T14:40:29.848436641Z Skipping ShutdownNow: child process not running
2020-07-29T14:40:29.848441151Z Waiting for child process to exit...
2020-07-29T14:40:29.848552551Z I0729 14:40:29.848471       1 trace.go:116] Trace[1298498081]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125 (started: 2020-07-29 14:39:59.849381215 +0000 UTC m=+0.005594313) (total time: 29.998886437s):
2020-07-29T14:40:29.848562191Z Trace[1298498081]: [29.998886437s] [29.998886437s] END
2020-07-29T14:40:29.848565627Z Pod Watch(<job-name-redacted>): done
2020-07-29T14:40:29.851615107Z panic: close of nil channel
2020-07-29T14:40:29.851622547Z 
2020-07-29T14:40:29.851624956Z goroutine 1 [running]:
2020-07-29T14:40:29.851627171Z github.com/karlkfi/kubexit/pkg/supervisor.(*Supervisor).Wait.func1(0xc00027a500)
2020-07-29T14:40:29.851629879Z 	/build/pkg/supervisor/supervisor.go:78 +0x54
2020-07-29T14:40:29.851632670Z github.com/karlkfi/kubexit/pkg/supervisor.(*Supervisor).Wait(0xc00027a500, 0x148b920, 0xc0001d4fc0)
2020-07-29T14:40:29.851635168Z 	/build/pkg/supervisor/supervisor.go:84 +0xc6
2020-07-29T14:40:29.851637312Z main.waitForChildExit(0xc00027a500, 0x0)
2020-07-29T14:40:29.851639549Z 	/build/cmd/kubexit/main.go:232 +0x32
2020-07-29T14:40:29.851651653Z main.fatalf(0xc00027a500, 0xc000278680, 0x13028e0, 0xa, 0xc0000b7f40, 0x1, 0x1)
2020-07-29T14:40:29.851655342Z 	/build/cmd/kubexit/main.go:261 +0xdc
2020-07-29T14:40:29.851658867Z main.main()
2020-07-29T14:40:29.851661713Z 	/build/cmd/kubexit/main.go:147 +0xaa0
```

This is quite unfortunate, as this means no tombstone is written and when the birth dependency comes up, it is never shutdown. This causes the job to hang forever, which is the reason we started using kubexit in the first place.

FWIW, I was able to easily reproduce just by running the client-server example on an empty kind cluster which didn't have default permissions for service accounts to list pods.

```
$ kubectl logs client-server-job-dq4c7 client
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
(1/9) Installing ncurses-terminfo-base (6.1_p20200118-r4)
(2/9) Installing ncurses-libs (6.1_p20200118-r4)
(3/9) Installing readline (8.0.1-r0)
(4/9) Installing bash (5.0.11-r1)
Executing bash-5.0.11-r1.post-install
(5/9) Installing ca-certificates (20191127-r2)
(6/9) Installing nghttp2-libs (1.40.0-r1)
(7/9) Installing libcurl (7.67.0-r0)
(8/9) Installing curl (7.67.0-r0)
(9/9) Installing tzdata (2020a-r0)
Executing busybox-1.31.1-r9.trigger
Executing ca-certificates-20191127-r2.trigger
OK: 13 MiB in 23 packages
Name: client
Graveyard: /graveyard
Tombstone: /graveyard/client
Birth Deps: server
Death Deps: N/A
Birth Timeout: 30s
Grace Period: 30s
Pod Name: client-server-job-dq4c7
Namespace: default
Watching pod updates...
E0729 23:41:15.074462      16 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Pod: pods "client-server-job-dq4c7" is forbidden: User "system:serviceaccount:default:default" cannot list resource "pods" in API group "" in the namespace "default"
E0729 23:41:15.075142      16 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Pod: pods "client-server-job-dq4c7" is forbidden: User "system:serviceaccount:default:default" cannot list resource "pods" in API group "" in the namespace "default"
E0729 23:41:18.181430      16 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Pod: pods "client-server-job-dq4c7" is forbidden: User "system:serviceaccount:default:default" cannot list resource "pods" in API group "" in the namespace "default"
E0729 23:41:23.509304      16 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Pod: pods "client-server-job-dq4c7" is forbidden: User "system:serviceaccount:default:default" cannot list resource "pods" in API group "" in the namespace "default"
E0729 23:41:32.711800      16 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Pod: pods "client-server-job-dq4c7" is forbidden: User "system:serviceaccount:default:default" cannot list resource "pods" in API group "" in the namespace "default"
Pod Watch(client-server-job-dq4c7): done
Error: timed out waiting for birth deps to be ready: 30s
Skipping ShutdownNow: child process not running
Waiting for child process to exit...
panic: close of nil channel

goroutine 1 [running]:
github.com/karlkfi/kubexit/pkg/supervisor.(*Supervisor).Wait.func1(0xc0002fa4c0)
        /build/pkg/supervisor/supervisor.go:78 +0x54
github.com/karlkfi/kubexit/pkg/supervisor.(*Supervisor).Wait(0xc0002fa4c0, 0x148b920, 0xc0004131c0)
        /build/pkg/supervisor/supervisor.go:84 +0xc6
main.waitForChildExit(0xc0002fa4c0, 0x0)
        /build/cmd/kubexit/main.go:232 +0x32
main.fatalf(0xc0002fa4c0, 0xc0002f6640, 0x13028e0, 0xa, 0xc0000a7f40, 0x1, 0x1)
        /build/cmd/kubexit/main.go:261 +0xdc
main.main()
        /build/cmd/kubexit/main.go:147 +0xaa0
```

--- a/pkg/supervisor/supervisor.go
+++ b/pkg/supervisor/supervisor.go
@@ -75,7 +75,9 @@ func (s *Supervisor) Start() error {
 func (s *Supervisor) Wait() error {
 	defer func() {
 		signal.Reset()
-		close(s.sigCh)
+		if s.sigCh != nil {
+			close(s.sigCh)
+		}
 		if s.shutdownTimer != nil {
 			s.shutdownTimer.Stop()
 		}

